lib/elixir/lib/option_parser.ex

```diff
@@ -492,4 +556,34 @@ defmodule OptionParser do
       :unknown
     end
   end
+
+  defp format_errors(errors, opts) do
+    details = Enum.map_join(errors, ". ", &format_error(&1, opts))
+    total = Enum.count(errors)
+    error_pluralize = pluralize(total, "error", "errors")
+    "#{total} #{error_pluralize} found! #{details}"
+  end
+
+  defp format_error({option, nil}, _) do
+    "Unknown option #{option}"
+  end
+
+  defp format_error({option, value}, opts) do
+    option_key =
+      String.lstrip(option, ?-)
+      |> String.to_atom()
+
+    type =
+      cond do
+        alias = opts[:aliases][option_key] ->
+          opts[:strict][alias]
+        true ->
+          opts[:strict][option_key]
+      end
```

Comment:
Personally, I'd replace this with a simple `if`. (also, I'd avoid using `alias`, because it's a reserved keyword and even if it works, it's usually highlighted by text editors)

```elixir
type =
  if opt_alias = opts[:aliases][option_key] ->
    opts[:strict][opt_alias]
  else
    opts[:strict][option_key]
  end
```

Wdyt?